functions/index.js

```diff
@@ -226,3 +242,35 @@ exports.addAllUsersFromCohortToDiscord = functions.https.onRequest(async (req, r
   })
   resp.send('OK')
 })
+
+exports.onUserCreated = functions.firestore
```

Comment:
Acho que seria melhor observar a criação de usuários no firestore users, pois lá tem todos os outros dados do usuário, não só id e email, faz sentido? Ou estou viajando?

Reply:
o que seria o firestore users ? Porque essa função tá ouvindo a criação de qualquer novo usuário no firestore de fato, não entendi bem a sua pergunta @danicuki. Na função eu busquei o o id e o email, mas poderia pegar qualquer dado do parametro snap, como discord, etc etc...
Como eu nao entendo muito de event drive (ainda, next conf do google ta ai pra isso, rs), criei esse modelo pra vc avaliar e validar, se fazia sentido esse raciocínio.

Reply:
essa função monitora a criação de usuários no `firebase authenticator`, não no firestore. Estou me referindo a criação de elementos na collection `users` do `firestore`. São duas coisas diferentes até onde entendo.

O correto seria algo tipo
```
exports.createUser = functions.firestore
    .document('users/{userId}')
    .onCreate((snap, context) => {
      // Get an object representing the document
      // e.g. {'name': 'Marie', 'age': 66}
      const newValue = snap.data();

      // access a particular field as you would any JS property
      const name = newValue.name;

      // perform desired operations ...
    });
```

Reply:
Dani, vc tá falando da função que eu criei na linha 246, correto ? Se for o caso, eu não vi diferença nenhuma além do nome da função. Ambas usam o onCreate, recebendo o snap e pegando os dados do usuário desse objeto(snap).
A minha e a sua, acessam o functions.firestore.document('users/{userId}').onCreate.

Perdão por não enxergar a diferença :/

Reply:
Nossa viajei. Foi mal Rômulo. Você está certo. A única coisa que falta então nesse caso é enviar o json com todos os dados do usuário, não só email e id. O melhor seria simplesmente mandar um campo userData com o json completo que chega no firestore.